### PR TITLE
Login Dialog Component - Replaced class by func. comp + added react-hook-form

### PR DESCRIPTION
**Pull request by priscilawebdev**
**Type:** Refactor

The following has been addressed in the PR: #116 

**Description:** 

in order to use react hooks, in this PR I updated the Login Dialog component by changing it from a class to a functional component.

I've also made it more pretty 💅

image

**Others things done in this PR:**

- I have turned off this lint rule: react/jsx-max-depth, because it is not really helpful. Please do you agree with me?

- I introduced this amazing library: [React-hook-form](https://react-hook-form.com/ ), in order to have a better form's state management + less verbose + better performance. The library is also very lightweight with only 5.7kB (minified + GZIPPED) 👏

- I installed `mutationobserver-shim` library to fix some tests issue.[ please see](https://react-hook-form.com/faqs/): 
image

### CODE REVIEW

**Comment by divyeshairmeet** [comment]
is this closed? Can I work on this?

File: .eslintrc [hunk 1]
```diff
@@ -83,7 +83,7 @@
     "react/jsx-indent": ["error", 2],
     "react/jsx-indent-props": ["error", 2],
     "react/jsx-key": ["error"],
-    "react/jsx-max-depth": ["error", { "max": 2}],
+    "react/jsx-max-depth":["error", { "max": 5}],
```
Comment (juanpicado) [inline .eslintrc]: 😆

**Comment by juanpicado** [comment]
Great job jere ! 👏

**Comment by priscilawebdev** [comment]
@juanpicado thanks 🙃🙂

**Pull request merged**